tars/util/gpool/gpool.go

```diff
@@ -0,0 +1,87 @@
+package gpool
+
+type Worker struct {
+	WorkerQueue chan *Worker
+	JobChannel  chan Job
+	Stop        chan struct{}
+}
+
+func (w *Worker) Start() {
+	go func() {
+		var job Job
+		for {
+			w.WorkerQueue <- w
```

Comment:
It is not like that.  The worker will first put itself to `WorkerQueue` when `Start`. And it will put it back when job is finished actually because of the loop.